consensus/spos/bls/subroundEndRound.go

```diff
@@ -155,6 +156,94 @@ func (sr *subroundEndRound) isBlockHeaderFinalInfoValid(cnsDta *consensus.Messag
 	return true
 }
 
+// receivedInvalidSignersInfo method is called when a message with invalid signers has been received
+func (sr *subroundEndRound) receivedInvalidSignersInfo(_ context.Context, cnsDta *consensus.Message) bool {
+	node := string(cnsDta.PubKey)
+
+	if !sr.IsConsensusDataSet() {
+		return false
+	}
+
+	if !sr.IsNodeLeaderInCurrentRound(node) { // is NOT this node leader in current round?
```

Comment:
No, it is ok, as the node variable holds the pubkey of the sender of this message and you want to check if this message was sent by the leader, otherwise you will decrease the peer honesty of this pub key which is not in charge to send this kind of info.

Reply:
For checking yourself there is a method named: IsSelfLeaderInCurrentRound() without any parameter

Reply:
right, missed the node variable setting to the message pubKey. 
variables need some renaming, e.g node to messageSender